packages/SwingSet/bin/rekernelize.js

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env -S node -r esm
+
+/* global require */
+
+import 'node-lmdb';
+import '@agoric/babel-standalone';
+import '@agoric/install-ses';
+
+import path from 'path';
+import process from 'process';
+
+import bundleSource from '@agoric/bundle-source';
+import { openLMDBSwingStore } from '@agoric/swing-store-lmdb';
+
+async function main() {
+  const argv = process.argv.slice(2);
+
+  let dbDir = '.';
+  if (argv.length > 1) {
+    console.error('usage: rekernelize [DBDIR]');
+    process.exit(1);
+  } else if (argv[0]) {
+    dbDir = argv[0];
+  }
+
+  const kernelStateDBDir = path.join(dbDir, 'swingset-kernel-state');
```

Comment:
I got this to work against a validator node, but I had to change this line, because the kernelDB is held in a different directory:

```
$ ls ~/.ag-chain-cosmos/data/ag-cosmos-chain-state/
data.mdb  lock.mdb  streams  streams.sqlite  xs-snapshots
```

Reply:
not sure what needs to be changed.. maybe remove this line and have the user point to one directory lower?

Reply:
I just stole the DB location logic from the `kerneldump` utility.  So now you can pass it the path to the `.mdb` file or to the directory containing the `.mdb` file or to a directory containing a canonical `swingset-kernel-state` directory and in all cases it will do the right thing.